src/stactools/viirs/cog.py

```diff
@@ -0,0 +1,158 @@
+
+from stactools.viirs.constants import MULTIPLE_NODATA
+from stactools.viirs.metadata import viirs_metadata
+
+
+def cogify(infile: str, outdir: str) -> List[str]:
+    """Creates COGs for the provided HDF5 file.
+
+    COGs are created using h5py as the data reader; rasterio and GDAL silently
+    convert int8 (signed byte) data to uint8 (byte) data.
+
+    Args:
+        infile (str): The input HDF5 file
+        outdir (str): The output directory
+
+    Returns:
+        List[str]: The COG hrefs
+    """
+    metadata = viirs_metadata(infile)
+    base_filename = os.path.splitext(os.path.basename(infile))[0]
+
+    all_keys: List[str] = []
+    with h5py.File(infile) as h5:
+        h5.visit(all_keys.append)
+        subdataset_keys = [
+            key
+            for key in all_keys
+            if isinstance(h5[key], h5py.Dataset) and "GRIDS" in key
+        ]
+
+    cog_paths = []
+    for subdataset_key in subdataset_keys:
+        sanitized_key = subdataset_key.replace(" ", "_")
+        rasterio_subdataset_path = f"HDF5:{infile}://{sanitized_key}"
+
+        parts = sanitized_key.split("/")
+        subdataset_name = parts[-1]
+        cog_filename = f"{base_filename}_{subdataset_name}.tif"
+        cog_path = os.path.join(outdir, cog_filename)
+
+        with warnings.catch_warnings():
+            warnings.simplefilter("ignore", category=NotGeoreferencedWarning)
+            with rasterio.open(rasterio_subdataset_path, "r") as tag_src:
+                rasterio_tags = tag_src.tags()
+
+        with h5py.File(infile) as h5:
+            data: Any = np.array(h5[subdataset_key])
+            if len(data.shape) == 1:  # skip single value (non-data) "grids"
+                continue
+            if len(data.shape) == 3:
+                raise ValueError(
+                    f"MultiBand COG creation not supported for "
+                    f"subdataset '{subdataset_name}' in product "
+                    f"'{metadata.product}'"
+                )
+
+            # gdal (and software built on gdal) doesn't always play well with signed byte data
+            data = np.int16(data) if data.dtype == "int8" else data
+
+            if "_FillValue" in h5[subdataset_key].attrs.keys():
+                temp = h5[subdataset_key].attrs["_FillValue"].item()
+            elif "_Fillvalue" in h5[subdataset_key].attrs.keys():
+                temp = h5[subdataset_key].attrs["_Fillvalue"].item()
+            nodata_values: List[Optional[int]] = [None] if temp == b"n/a" else [temp]
+
+            nodata_new: int
+            if metadata.product in MULTIPLE_NODATA:
+                subdatasets = MULTIPLE_NODATA[metadata.product]
+                if subdataset_name in subdatasets:
+                    nodata_values = cast(
+                        List[Optional[int]], subdatasets[subdataset_name]["multiple"]
+                    )
+                    nodata_new = cast(int, subdatasets[subdataset_name]["new"])
+
+            if len(nodata_values) > 1:
```

Comment:
Can these `if` statements be collapsed? In other words, is there a case when `metadata.product in MULTIPLE_NODATA` but `len(nodata_values) == 1`?

Reply:
Yep. Collapsed and simplified